onnxruntime/core/providers/webnn/builders/impl/expand_op_builder.cc

```diff
@@ -88,6 +88,10 @@ bool ExpandOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers
     LOGS(logger, VERBOSE) << "Cannot get shape.";
     return false;
   }
+  if (std::any_of(new_shape.begin(), new_shape.end(), [](int64_t dimension) { return dimension == 0; })) {
+    LOGS(logger, VERBOSE) << "Expand does not support new shape with 0 dimension.";
+    return false;
+  }
```

Comment:
Could you also fix the `reshape_op_builder.cc` ?

Reply:
It's already check in `reshape_op_builder.cc`.